src/app/search/autocomplete.test.ts

```diff
@@ -24,49 +25,99 @@ describe('autocompleteTermSuggestions', () => {
+    ['is:haspower is:b', 'is:haspower is:bow'],
+    ['(is:blue ju|n)', '(is:blue tag:junk)'],
+    ['is:bow is:v|oid', 'is:bow is:void'],
+    // TODO: range overloads should really work
+    ['season:>outl', 'Expected failure'],
+    ['not(', 'Expected failure'],
   ];
 
-  test.each(cases)('autocomplete within query for {%s}', (queryWithCaret) => {
-    const [caretIndex, query] = extractCaret(queryWithCaret);
-    const candidates = autocompleteTermSuggestions(query, caretIndex, filterComplete, searchConfig);
-    expect(candidates).toMatchSnapshot();
-  });
+  test.each(cases)(
+    'autocomplete within query for {%s}',
+    (queryWithCaret: string, expected: string) => {
+      const [caretIndex, query] = extractCaret(queryWithCaret);
+      const candidates = autocompleteTermSuggestions(
+        query,
+        caretIndex,
+        filterComplete,
+        searchConfig
+      );
+      expect(candidates[0]?.query.body ?? 'Expected failure').toBe(expected);
+    }
+  );
+
+  const multiWordCases: [query: string, expected: string][] = [
+    ['arctic haz', 'name:"arctic haze"'],
+    ['is:weapon arctic haz| -is:exotic', 'is:weapon name:"arctic haze" -is:exotic'],
+    ['name:"arctic haz', 'name:"arctic haze"'],
+    ["name:'arctic haz", 'name:"arctic haze"'],
+    ['name:"foo" arctic haz', 'name:"foo" name:"arctic haze"'],
+    ["ager's sce", 'name:"ager\'s scepter"'],
+    ['the last word', 'name:"the last word"'],
+    ['acd/0 fee', 'name:"acd/0 feedback fence"'],
+    ['stat:rpm:200 first in, last', 'stat:rpm:200 name:"first in, last out"'],
+    ['two-tail', 'name:"two-tailed fox"'],
+    ['(is:a or is:b) and (is:c or multi w|)', '(is:a or is:b) and (is:c or name:"multi word")'],
+    ['"rare curio" arctic haz', '"rare curio" name:"arctic haze"'],
+    ['"rare curio" or arctic haz', '"rare curio" or name:"arctic haze"'],
+    ['toil and trou', 'name:"toil and trouble"'],
+    ['perkname:"fate of', 'perkname:"fate of all fools"'],
+    ['perkname:fate of', 'perkname:"fate of all fools"'],
+    // Expected (or at least not yet supported) failures:
+    ['rare curio or arctic haz', 'No fake autocomplete for: rare curio or arctic haz'], // no way to know that isn't one name
+    ['name:heritage arctic haze', 'No fake autocomplete for: name:heritage arctic haze'], // this actually works in the app but relies on the full manifest
+  ];
 
-  const multiWordCases: [query: string, mockCandidate: string][] = [
-    ['arctic haz', 'arctic haze'],
-    ['is:weapon arctic haz| -is:exotic', 'arctic haze'],
-    ['name:"foo" arctic haz', 'arctic haze'],
-    ["ager's sce", "ager's scepter"],
-    ['the last word', 'the last word'],
-    ['acd/0 fee', 'acd/0 feedback fence'],
-    ['stat:rpm:200 first in, last', 'first in, last out'],
-    ['two-tail', 'two-tailed fox'],
-    ['(is:a or is:b) and (is:c or multi w|)', 'multi word'],
-    ['arctic  haz', 'arctic haze'], // two spaces inbetween words
```

Comment:
This is the only regression I know of - we no longer collapse redundant whitespace. This is something we could add back in at the suggestion generator if we wanted.